### PR TITLE
Updated incident transform and load scripts

### LINKED ISSUE #95: [Update] Update the incidents transform script
## Task Description
- Update the transform script to handle null values more effectively when parsing xml data, to remove description as we have chosen to remove this data from the ERD (as per PR #90) and it is not being used for the alerts (the summary is sufficient)
## User Stories
- As an engineer I want to update the script to optimise it so that it doesn't involve processing any unnecessary data and also so that any potential null values that might come through are handled gracefully to avoid causing problems with notification creation and database insertion. 
## Relevant Files [If Available]
- `transform.py`

### LINKED ISSUE #96: [Update][Bug] Update incidents load script
## Task Description
- Update the incidents `load.py` script to account for errors when loading live data into the database. I need to account for multiple operators being affected by the same incident and also for the possibility that the operator affected might not already exist in the `operators` table in the database (this currently causes the script to skip the insertion for that particular message)
## User Stories
- As an engineer I want to make sure that I account for errors when loading to the database so that errors aren't thrown/messages are not skipped.
- As an engineer I want to log the events as they are happening so that it is easier to debug problems.
## Relevant Files [If Available]
- `load.py`

### PR DESCRIPTION
**Pull request by SaniyaShaikhh**
## Updated Transform Script 
- The `transform.py` functions have been broken down a bit more for better code readability
- Description has been removed
- Cleared variable has been added (True if an incident has been resolved, false if not)

## Updated Load Script
For `load.py`:
- I have added a function which executes the `UPDATE_OR_INSERT_QUERY`, this either updates or adds a new row to the database depending on if that incident uuid already exists. 
- I have also added the `add_operator_id` function which adds an operator to the operators table in the database if it doesn't already exist.

closes #95 
closes #96

### CODE REVIEW

**Comment by SaniyaShaikhh** [comment]
As I have edited the `transform.py` script, the `test_transform.py` file needs to be updated to pass the tests.

File: incidents/load.py [hunk 1]
```diff
@@ -18,45 +46,61 @@ def get_db_connection(config: dict[str, str]) -> connection:
...
+    incident_uuid = data["incident_number"]
 
-    operator_id = get_operator_id(config, data)
-    values = (operator_id,) + get_values_for_insertion(data)
-    insert_into_db(config, values)
+    for operator_code, operator_name in operators.items():
+        operator_id = get_operator_id(config, operator_code, operator_name)
+        insert_values = (
+            (operator_id,) + get_values_for_insertion(data) + (operator_id, incident_uuid))*2
+        insert_into_db(config, insert_values)
```
Comment (isaacschaessenscoleman) [inline incidents/load.py:106]: Nicely done for loop.

File: incidents/load.py [hunk 2]
```diff
@@ -18,45 +46,61 @@ def get_db_connection(config: dict[str, str]) -> connection:
...
+    conn = get_db_connection(config)
+    with conn.cursor() as cur:
+        cur.execute("""INSERT INTO operators ("operator_name", "operator_code")
+        VALUES (%s, %s)
+        RETURNING operator_id;""", (operator_name, operator_code))
+        conn.commit()
+        op_id = cur.fetchone()
         conn.close()
-        return int(op_id)
+        return int(op_id["operator_id"])
```
Comment (isaacschaessenscoleman) [inline incidents/load.py:93]: New function looks good.

File: incidents/transform.py [hunk 3]
```diff
@@ -1,90 +1,111 @@
...
+    operator_names = [operator_name.text for operator_name in root.findall(
+        ".//ns:Affects/ns:Operators/ns:AffectedOperator/ns:OperatorName", namespaces)]
+    return operator_names
+
+
+def process_xml(xml_message: str, namespaces: dict) -> dict:
+    """Processes each XML message to extract relevant information."""
+
+    if not xml_message:
+        return None
```
Comment (isaacschaessenscoleman) [inline incidents/transform.py:42]: Good edge case handling.

**Review by isaacschaessenscoleman — APPROVED** [review]
Transform file looks a lot better with refactored, clean and easy-to-read functions.

For the load script, there are really good and clear solutions to handling different queries based on certain conditions.

**Comment by SaniyaShaikhh** [comment]
I have opened an issue #112 to update `test_transform.py` so that it works for the updated transform script

**Pull request merged**